### ISSUE TITLE
Building acs-engine is extremely slow in Docker container

### ISSUE THREAD

**Issue by wbuchwalter** [Azure/acs-engine#1101]
---

**Is this an ISSUE or FEATURE REQUEST?**  Issue

---

**What version of acs-engine?**: latest

---

Building acs-engine is currently extremely slow, at least on Mac. 
Specifically, `make build` calls `go list ./…` which takes forever before listing (still hanging after 10 minutes in my case).

**Comment by seanknox** [comment]
Hey @wbuchwalter, a few questions:
- What version of acs-engine?
- What version of MacOS?
- Anything obvious stick out in Activity Monitor?

On a freshly cloned acs-engine, `make bootstrap build` completes quickly on my machine:
```
$ time make bootstrap build
...
real	0m14.379s
user	0m19.561s
sys	0m8.119s
```

**Comment by wbuchwalter** [comment]
Hey @seanknox,

* I did a fresh clone this morning, commit `05871f14`, still seeing the issue.
*  Sierra (`10.12.5`)
* Nothing really. I'm building inside a container (`./script/devenv.sh`), nothing obvious is sticking out, the `go` process is at around 10% CPU usage.

I am working with 2 other people, everyone is on Mac OS and we are all seeing this issue.

Calling `go list ./...` by itself is sufficient to reproduce the issue in my case.

**Comment by seanknox** [comment]
Talking to @wbuchwalter, the issue happens when running `go list ./...` inside an acs-engine container on a Mac. [Docker for Mac is known to have an issue with extremely slow volumes](https://github.com/docker/for-mac/issues/77). However, William said that as of ~20 days ago this worked ok, so we'll need to dig into the regression.

**Comment by wbuchwalter** [comment]
After doing some splicing, The issue appears to have been introduced by commit https://github.com/Azure/acs-engine/commit/8a47cbd8e8ecbe1f9e0372529b621b66718ff374

**Comment by seanknox** [comment]
This issue is caused by
1) slow file volumes on Docker for Mac when there are many files in the volume
2) growth of dev dependencies in `vendor/` (especially azure-sdk-for-go`)

Proposed fix in #1206
